### PR TITLE
[evm] enable Merge at Redsea height

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
at title.

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [x] fullsync
- [] Other test (please specify)

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -201,6 +206,6 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
-replace github.com/ethereum/go-ethereum => github.com/iotexproject/go-ethereum v0.4.2
+replace github.com/ethereum/go-ethereum => github.com/iotexproject/go-ethereum v1.7.4-0.20231009175652-26939171068a
```
Comment (dustinxie) [inline go.mod:209]: after full-sync test, this will be given a tag and updated

File: go.mod [hunk 2]
```diff
@@ -4,13 +4,13 @@ go 1.18
 
 require (
 	github.com/cenkalti/backoff v2.2.1+incompatible
```
Comment (envestcc) [inline go.mod:6]: are all the module upgrades required?

**Review by envestcc — APPROVED** [review]
Please also give the activated EIPs list from London to Merge

**Review by CoderZhi — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (dustinxie) [reply go.mod:6]: this is the result of running `go mod tidy`

**Pull request merged**